addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyMissingScanRule.java

```diff
@@ -146,4 +96,61 @@ private String getAlertAttribute(String key) {
     public Map<String, String> getAlertTags() {
         return ALERT_TAGS;
     }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return Arrays.asList(
+                alertMissingCspHeader().build(),
+                alertObsoleteCspHeader().build(),
+                alertObsoleteCspHeader().build());
+    }
+
+    private static boolean hasCspHeader(HttpMessage msg) {
+        return !msg.getResponseHeader()
+                .getHeaderValues(HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                .isEmpty();
+    }
+
+    private static boolean hasObsoleteCspHeader(HttpMessage msg) {
+        return !msg.getResponseHeader().getHeaderValues("X-Content-Security-Policy").isEmpty()
+                || !msg.getResponseHeader().getHeaderValues("X-WebKit-CSP").isEmpty();
+    }
+
+    private static boolean hasCspReportOnlyHeader(HttpMessage msg) {
+        return !msg.getResponseHeader()
+                .getHeaderValues("Content-Security-Policy-Report-Only")
+                .isEmpty();
+    }
+
+    private AlertBuilder buildAlert(int risk) {
+        return newAlert()
+                .setRisk(risk)
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setCweId(693) // CWE-693: Protection Mechanism Failure
+                .setWascId(15); // WASC-15: Application Misconfiguration
+    }
+
+    private AlertBuilder alertMissingCspHeader() {
+        return buildAlert(Alert.RISK_MEDIUM)
+                .setName(getAlertAttribute("name"))
+                .setDescription(getAlertAttribute("desc"))
+                .setSolution(getAlertAttribute("soln"))
```

Comment:
We seem to only have one solution so it can be done in `buildAlert(int risk)`

Reply:
Same could be done for "refs" and let it be overridden by `alertCspReportOnlyHeader`, which is the only one different.